File: models/cssdandsterilization/SterilizationCycle.js
// models/CSSD/SterilizationCycle.js
import mongoose from "mongoose";

const SterilizationCycleSchema = new mongoose.Schema({
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },
  batchId: { type: mongoose.Schema.Types.ObjectId, ref: "InstrumentBatch", required: true },
  cycleNumber: { type: Number, required: true },
  startTime: { type: Date, default: Date.now },
  endTime: Date,
  performedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  machineUsed: String,
  status: { type: String, enum: ["Pending", "Completed", "Failed"], default: "Pending" }
}, { timestamps: true });

export default mongoose.model("SterilizationCycle", SterilizationCycleSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/SterilizationCycle.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "sterilization_cycles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SterilizationCycle extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "batch_id", nullable = false)
    private InstrumentBatch batch;

    @Column(name = "cycle_number", nullable = false)
    private Integer cycleNumber;

    @Column(name = "start_time")
    private LocalDateTime startTime = LocalDateTime.now();

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "performed_by")
    private User performedBy;

    @Column(name = "machine_used")
    private String machineUsed;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Pending;

    public enum Status {
        Pending, Completed, Failed
    }
}
